### PR TITLE
Add no-info and quiet options to machine init

### PR DESCRIPTION
**Pull request by cblecker**
#### Does this PR introduce a user-facing change?
```release-note
Add --no-info and --quiet options to podman machine init, to align with flags from podman machine start
```
[NO NEW TESTS NEEDED]

### CODE REVIEW

**Comment by rhatdan** [comment]
@cblecker need
[NO NEW TESTS NEEDED]

Added to the PR to pass tests.

@ashley-cui @baude PTAL
LGTM

**Review by ashley-cui — COMMENTED** [review]
LGTM

**Review by Luap99 — COMMENTED** [review]
This seems to only cover cases where it writes to stdout and we already have the `--noout` option for that. I would argue therefore the flag is not necessary.

And in case this is needed you need to cover all the println's in the backend code as well. I haven't looked at the full code path but I found at least this here
https://github.com/containers/podman/blob/8a342404e6ac5c4e632e3cb273a665a6cf81bc90/pkg/machine/qemu/machine.go#L363
And when using --now it has to pass the options to the start code as well to make it not print as well.

**Comment by cblecker** [comment]
@Luap99 The thing that prompted me to open this PR was the fact that the options were not properly validating when using `--now`.. and when you tried to do that, the hint to start it after creation is still passed to stdout

**Comment by cblecker** [comment]
Tests are now passing

**Comment by vrothberg** [comment]
@Luap99 can you take another look?

**Comment by TomSweeneyRedHat** [comment]
LGTM
but would like a head nod from @Luap99

File: cmd/podman/machine/init.go [hunk 1]
```diff
@@ -118,6 +124,8 @@ func initMachine(cmd *cobra.Command, args []string) error {
 		vm  machine.VM
 	)
 
+	initOpts.NoInfo = initOpts.Quiet || initOpts.NoInfo
```
Comment (Luap99) [inline cmd/podman/machine/init.go:127]: where are initOpts.Quiet or initOpts.NoInfo set, they are never assigned anywhere I can see. The flags use startOpts which seems wrong.

**Review by Luap99 — CHANGES_REQUESTED** [review]
This does not work right now.

Also you have to touch the backend code, I assume that quiet must suppress the download bar as well as the other output, this is the current output with your change:
```
Downloading VM image: fedora-coreos-37.20230218.2.0-qemu.x86_64.qcow2.xz: done  
Extracting compressed file
Image resized.
Machine init complete
To start your machine run:

        podman machine start

```

**Comment by rhatdan** [comment]
@cblecker @Luap99 @ashley-cui what should be done with this PR?

**Comment by baude** [comment]
i always liked the --quiet --quite or -v -v for verbose

**Comment by vrothberg** [comment]
> i always liked the --quiet --quite or -v -v for verbose

@baude I share that preference but `podman machine start` already uses `--no-info`.

**Comment by ashley-cui** [comment]
--no-info does something different than quiet  - no-info suppresses additional information about actions that the user may want to do to enhance their machine experience, while quiet suppresses that info, as well as any status information, waiting for the VM to boot etc. I'd be okay with aliasing no-info to something else, but I believe in the original issue the no-info behavior was specifically asked for.

**Comment by ashley-cui** [comment]
I still think --quiet and --no-info are two different things, especially that the initial request was for --no-info and not --quiet. I'd be okay with aliasing, or something like -q 0 and -q 1 where -q 1 is no-info or something like that, but i think the two behaviors have to be distinct.

**Comment by vrothberg** [comment]
Closing this PR then.  Let's first iron out what we want/need.  Then we can reopen or create a new one.

**Pull request closed without merging**